rfc/development-cycle.md

```diff
@@ -0,0 +1,180 @@
+
+
+**alpha, beta, rc**: Each release goes through some optional number of alpha, beta, and RC release, before the final. For some final release `{x}.{y}.{z}` it can be modeled with this state machine
+
+1. Start by releasing one of these:
+    1. `{x}.{y}.{z}a0`
+    2. `{x}.{y}.{z}b0`
+    3. `{x}.{y}.{z}rc0`
+    4. `{x}.{y}.{z}`
+2. Then the next release can be either an increment of the last number or a release of one of the initial following your
+   release. For example after `{x}.{y}.{z}b10` you could release `{x}.{y}.{z}b11` or `{x}.{y}.{z}rc0` or `{x}.{y}.{z}` (final).
+
+**Previous Ancestor**: The next increment will be the git ancestor of the previous increment.
+
+* Patch: `{x}.{y}.{z}` is an ancestor of `{x}.{y}.{z+1}`.
+* Minor: `{x}.{y}.0` is an ancestor of `{x}.{y+1}.0`.
+* Major: `{x}.0.0` is an ancestor of `{y+1}.0.0`.
+
+**Single Active**: No commits will be queued up for a release until the previous increment was fully release
+(this also falls out of the above assumption). 
+
+* Patch: No commits towards `{x}.{y}.{z}` till `{x}.{y}.{z-1}` final is released.
+* Minor: No commits towards `{x}.{y}.0` till `{x}.{y-1}.0` final is released.
+* Major: No commits towards `{x}.0.0` till `{x-1}.0.0` final is released.
+
+The last two mean you have to totally finish release before you start on the next one. For example, it would be illegal to have releases in this order: `1.2.0a0`, `1.3.0a0`, `1.2.0`, then `1.3.0`.
+
+**JavaScript versions In Sync** We are ok always keeping the JS version bumps in sync. Meaning that if we do a major release of the Python package we also do a major release of all JS versions.
+
+## Strategy
+
+Based on these assumption we can create a consistant branching strategy.
+
+### Branches
+
+Each branch should be aligned to different releases:
+
+1. Patch: `{x}.{y}.x` branches.
+2. Minor: `{x}.x.0` branches. After final release `{x}.{y}.0` create new patch `{x}.{y}.x` branch.
+3. Major: `master` branch. After release `{x}.0.0` create new minor `{x}.x.0` branch and new patch `{x}.0.x` branch.
+
+Each increment has a corresponding "latest" branch for the greatest working version. For example with two branches `2.x.0`, `1.x.0`  the "latest" minor branch is `2.x.0`. The latest major branch is always `master`.
+
+*Note: We could instead have separate branches for each version, but this isn't necessarily if we are going by the **Single Active** assumption above.*
+
+
+### PR Semver Tags
+
+Each PR should be tagged with one of three tags `semver:patch`, `semver:minor`, `semver:major` before it is merged. We could do this manually or by possibly giving a first guess [using type analysis](https://api-extractor.com/) or some commit message keywords. We should have a bot that blocks merging if one of these is not added.
```

Comment:
:100: for the bot idea.  We need to be more liberal with the triage rights as well.